### PR TITLE
Micah dev

### PR DESCRIPTION
**Pull request by Ixw123**
Measurement:area implemented, yet to be tested or well documented. Let me know what you would like me to change or just change it or comment on this or in the code yourself. Thanks

### CODE REVIEW

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            p2 = coords[middleIndex]
+            p3 = coords[upperIndex]
+            total += (p3[0].toRadians() - p1[0].toRadians()) * sin(p2[1].toRadians())
+        }   
+        total *= (EARTH_RADIUS * EARTH_RADIUS / 2)
+    } 
+    return total
+}
+//private fun area()
+fun Geometry.area() = calculateArea(this)
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Provide the return type explicitly: `fun Geometry.area(): Double`

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+fun Polygon.area() = calculateArea(this)
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove, this is handled by `fun Geometry.area(): Double`

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+fun MultiPolygon.area() = calculateArea(this)
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove this is handled by `fun Geometry.area(): Double`

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+fun FeatureCollection.area(): Double {
+    var total = 0.0
+    for (feature in features) {
+        total += feature.area()
+    }
+    return total
+}
+
+//fun main() {
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+// For testing purposes
+//val EARTHRADIUS = 6378137
+
+private fun calculateArea( geom: Geometry): Double{
+    var total = 0.0
+    //var i: Int
+    when (geom) {
+        is Polygon -> { 
+            //val size = geom.coordinates.size
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            return polygonArea(geom.coordinates)
+        }
+        is MultiPolygon -> {
+            //val size = geom.coordinates.size
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            for (i in 0..(geom.coordinates.size - 1)) {
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Use `for(i in 0 until geom.coordinates.size`) or `for(i in geom.coordinates.indices)`

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        is MultiPoint -> return 0.0
+        is LineString -> return 0.0
+        is MultiLineString -> return 0.0
+        else -> throw UnsupportedOperationException("Can not calculate area of unrecognized Geometry type: ${geom::class.java.name}")
+    }
+}
+
+private fun polygonArea(coordinates: Array<Array<Position>>): Double {
+    var total = 0.0
+    //val coordSize = coordinates.size
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Remove

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    if (coordinates.isNotEmpty()) {
+        total += abs(ringArea(coordinates[0]))
+        for (i: Int in 1..(coordinates.size - 1)){
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Type declaration on `i` is not needed. Use `for(i in 1 until coordinates.size)`

File: src/main/kotlin/org/turfkt/Measurement.kt [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    var lowerIndex: Int
+    var middleIndex: Int
+    var upperIndex: Int
+    //var i: Int
+    var total = 0.0
+
+    val coordinateLength = coords.size
+
+    if (coordinateLength > 2) {
+        for (i in 0..(coordinateLength - 1)) {
```
Comment (hudsonb) [inline src/main/kotlin/org/turfkt/Measurement.kt]: Use `until`

**Review by hudsonb — CHANGES_REQUESTED** [review]

**Review by hudsonb — CHANGES_REQUESTED** [review]
The changes to .gitignore should not be included, otherwise looks good.

Can you back out the changes to .gitignore?

**Comment by Ixw123** [comment]
I can definitly tru I'm not quite sure how to do that

**Comment by hudsonb** [comment]
You can just revert the changes and commit.

I'd rebase the branch and edit the commit that added the modifications, but
that'll probably take you awhile to do if you aren't familar with the
process. I'm going to squash the branch before committing anyways.

**Pull request merged**